### ISSUE TITLE
MOSS-TTS-Local 微调是否支持断点续训？

### ISSUE THREAD

**Issue by freenewfree** [OpenMOSS/MOSS-TTS#181]
尊敬的MOSS-TTS项目组：
        我在用贵项目的MOSS-TTS-Local进行微调，设置了8轮训练，中间停止，再开始训练，每次都是从头开始训练，而不能接着上次断点轮数继续训练，请问该项目支持断点续训吗？如果有是要加什么参数？
        感谢你们开源如此好的项目！！

**Comment by alpacaking** [comment]
您好，目前 `moss_tts_local/finetuning/sft.py` 暂时不支持严格意义上的断点续训，也就是不会自动恢复 optimizer、lr scheduler、global_step、epoch、dataloader/RNG 等训练状态，所以重新执行原命令会从第 0 个 epoch 重新开始。

当前脚本会在每个 epoch 后保存 `checkpoint-epoch-*`，这些 checkpoint 可以直接用于推理，也可以作为新的初始化权重继续训练。例如可以把 `--model-path` 指向上一次保存的 checkpoint：

accelerate launch moss_tts_local/finetuning/sft.py \
    --model-path output/moss_tts_local_sft/checkpoint-epoch-X \
    --train-jsonl train_with_codes.jsonl \
    --output-dir output/moss_tts_local_sft_continue \
    ...其他训练参数...

这种方式会从已微调后的模型权重继续训练，但 optimizer 和学习率调度会重新初始化，因此不是完全等价的 resume。

**Comment by freenewfree** [comment]
> 您好，目前 `moss_tts_local/finetuning/sft.py` 暂时不支持严格意义上的断点续训，也就是不会自动恢复 optimizer、lr scheduler、global_step、epoch、dataloader/RNG 等训练状态，所以重新执行原命令会从第 0 个 epoch 重新开始。
> 
> 当前脚本会在每个 epoch 后保存 `checkpoint-epoch-*`，这些 checkpoint 可以直接用于推理，也可以作为新的初始化权重继续训练。例如可以把 `--model-path` 指向上一次保存的 checkpoint：
> 
> accelerate launch moss_tts_local/finetuning/sft.py --model-path output/moss_tts_local_sft/checkpoint-epoch-X --train-jsonl train_with_codes.jsonl --output-dir output/moss_tts_local_sft_continue ...其他训练参数...
> 
> 这种方式会从已微调后的模型权重继续训练，但 optimizer 和学习率调度会重新初始化，因此不是完全等价的 resume。

您好非常感激的您的回复，知道了。再次表示感谢！谢谢你们的辛苦付出！
